common-libs/common/src/shared/DidResolver.ts

```diff
@@ -0,0 +1,57 @@
+import { RegistryApiService } from '@affinidi/internal-api-clients'
+
+type CachedPromises = Map<string, Promise<Record<string, any>>>
+type ServiceWithCache = {
+  service: RegistryApiService
+  cache: CachedPromises
+}
+
+type ConstructorOptions = ConstructorParameters<typeof RegistryApiService>[0]
+
+const services: Map<string, ServiceWithCache> = new Map()
+
+const getService = (options: ConstructorOptions) => {
+  const { accessApiKey, registryUrl } = options
+  const cacheKey = JSON.stringify({ accessApiKey, registryUrl })
+  if (!services.has(cacheKey)) {
+    services.set(cacheKey, {
+      service: new RegistryApiService(options),
+      cache: new Map(),
```

Comment:
We already have a limited version of this in `Affinidi`.

If needed, we could also make a wrapper around `DidResolver` with a pre-populated cache. We cannot do this in `DidResolver` itself because pre-populated cache cannot be trusted and therefore cannot be reused between users (and `DidResolver` _is_ reused between users)